CRM/Event/Page/EventInfo.php

```diff
@@ -271,17 +268,25 @@ public function run() {
     $this->assign('registerClosed', !empty($values['event']['is_online_registration']) && !$isEventOpenForRegistration && CRM_Core_Permission::check('register for events'));
     $this->assign('allowRegistration', $allowRegistration);
 
-    $session = CRM_Core_Session::singleton();
-    $params = [
-      'contact_id' => $session->get('userID'),
-      'event_id' => $values['event']['id'] ?? NULL,
-      'role_id' => $values['event']['default_role_id'] ?? NULL,
-    ];
-
-    if (($this->isEventFull() && $noFullMsg === 'false') || CRM_Event_BAO_Event::checkRegistration($params)) {
-      $statusMessage = $this->getEventValue('event_full_text');
-      if (CRM_Event_BAO_Event::checkRegistration($params)) {
-        if ($noFullMsg == 'false') {
+    $isAlreadyRegistered = $this->isAlreadyRegistered();
+    // noFullMsg was originally passed in to suppress the message about the event being full. The intent
+    // was originally such that when you were sending the user back to the info page after registering
+    // they would not be told it was full. Along the way it got overloaded to encompass
```

Comment:
Aaaah. :bulb:  I was definitely scratching my head to understand that parameter. Comment helps a lot.